openhexa/sdk/workspaces/current_workspace.py

```diff
@@ -147,8 +153,97 @@ def tmp_path(self) -> str:
         # We can remove this once we deprecate this way of running pipelines
         return os.environ["WORKSPACE_TMP_PATH"] if "WORKSPACE_TMP_PATH" in os.environ else "/home/hexa/tmp"
 
-    @staticmethod
-    def dhis2_connection(identifier: str = None, slug: str = None) -> DHIS2Connection:
+    def get_connection(
+        self, identifier: str
+    ) -> typing.Union[
+        DHIS2Connection,
+        PostgreSQLConnection,
+        IASOConnection,
+        S3Connection,
+        S3Connection,
+        CustomConnection,
+        None,
+    ]:
+        """Get a connection by its identifier.
+
+        Parameters
+        ----------
+        identifier : str
+            The identifier of the connection in the OpenHEXA backend
+
+        Returns
+        -------
+        Connection
+            The connection
+
+        Raises
+        ------
+        ValueError
+            If the connection does not exist
+        """
+        fields = {}
+        connection_type = None
+        if self._connected:
+            response = graphql(
+                """
+                query getConnection($slug: String!) {
+                    connectionBySlug(slug: $slug) {
+                        type
+                        fields {
+                            code
+                            value
+                        }
+                    }
+                }
+            """,
+                {"slug": identifier},
+            )
+            data = response["connectionBySlug"]
+            if data is None:
+                raise ValueError(f"Connection {identifier} does not exist.")
+
+            for d in data["fields"]:
+                fields[d.get("code")] = d.get("value")
+
+            connection_type = data["type"]
+        else:
+            try:
+                env_variable_prefix = stringcase.constcase(identifier.lower())
+                for key, val in os.environ.items():
+                    if key.startswith(env_variable_prefix) and key != env_variable_prefix:
```

Comment:
Well, why don't you just do a `key.startswith(f"{env_variable_prefix}_")` ?

Reply:
Bien vu!